core/vm_env.go

```diff
@@ -41,29 +41,38 @@ func GetHashFn(ref common.Hash, chain *BlockChain) func(n uint64) common.Hash {
 }
 
 type VMEnv struct {
-	state  *state.StateDB
+	// State to use for executing
+	state *state.StateDB
+	// Header information
 	header *types.Header
-	msg    Message
-	depth  int
-	chain  *BlockChain
-	typ    vm.Type
-
-	getHashFn func(uint64) common.Hash
-	// structured logging
+	// Message appliod
```

Comment:
Could you move the comments after the fields? Looks weird this way :P Also `appliod`